API/Controllers/WizardController.cs

```diff
@@ -0,0 +1,91 @@
+* 
+* This program is free software: you can redistribute it and/or modify 
+* it under the terms of the GNU Lesser General Public License as published 
+* by the Free Software Foundation version 3 of the License.
+* 
+* This program is distributed in the hope that it will be useful, 
+* but WITHOUT ANY WARRANTY; without even the implied warranty 
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+* See the GNU Lesser General Public License for more details.
+* 
+* You can find a copy of the GNU Lesser General Public License 
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using API.Resources;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Models;
+using Services.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API.Controllers
+{
+
+    /// <summary>
+    ///     The controller that handles search requests
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class WizardController : ControllerBase
+    {
+
+        private readonly SourceManagerService sourceManagerService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WizardController"/> class.
+        /// </summary>
+        /// <param name="mapper">The mapper.</param>
+        /// <param name="sourceManagerService">The source manager service.</param>
+        public WizardController(IMapper mapper, SourceManagerService sourceManagerService)
+        {
+            this.sourceManagerService = sourceManagerService;
+
+        }
+
+        /// <summary>
+        /// Gets the wizard information.
+        /// </summary>
+        /// <param name="sourceURI">The source URI.</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Authorize]
+        public async Task<IActionResult> GetWizardInformation(Uri sourceURI)
+        {
+
+            if(sourceURI == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Source uri is null or empty.",
+                    Detail = "the incoming source url is not valid.",
+                    Instance = "6D63D9FA-91D6-42D5-9ACB-461FBEB0D2ED"
+                };
+                return BadRequest(problem);
+            }
+            Project project = sourceManagerService.FetchProject(sourceURI);
+            if(project == null)
+            {
+                return NotFound();
+            }
+            if(project.Name == null && project.ShortDescription == null && project.Description == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Project not found.",
+                    Detail = "the incoming source url aims at a gitlab which is either not instanciated or is a group.",
```

Comment:
I'm confused with the detail error message, what precisely do you want to tell the user with this error? Especially with the part "or is a group"?

Reply:
The normal schema of gitlab is:
gitlab.com/owner/project
gitlab.com/group/project

but when a user gives an url to a group like:
gitlab.com/group/subgroup
it still resolves with the api but it does not give data back(as expected)

But also when you create a project on gitlab it gives something along the lines of:
get started with gitlab make your first  commit.

and if you havent done that then the api cannot get any information.
referring to the not instanciated part.

Reply:
That makes more sense, thanks for the explanation! 
To someone who has not worked with Gitlab much (API), this might not be clear. Should we document this somewhere in your opinion?

Reply:
i put the information here:
https://github.com/DigitalExcellence/dex-backend/wiki/Specific-error-details